submissions/ses3332002/tiny-js-world/index.js

```diff
@@ -0,0 +1,105 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+const dog = {
+  species: "dog",
+  name: "Toby",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Woof-woof!",
+  friends: [
+    "Puss"
+  ]
+};
+
+const cat = {
+  species: "cat",
+  name: "Puss",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Meow!",
+  friends: [
+    "Toby",
+    "Halley"
+  ]
+};
+
+const man = {
+  species: "human",
+  name: "Jack",
+  gender: "male",
+  legs: 2,
+  hands: 2,
+  saying: "Hey baby!",
+  friends: [
+    "Jennie",
+    "Halley"
+  ]
+};
+
+const woman = {
+  species: "human",
+  name: "Jennie",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: "Hi darling!",
+  friends: [
+    "Jack",
+    "Toby"
+  ]
+};
+
+const catWoman = {
+  species: "super-hero",
+  name: "Halley",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: cat.saying,
+  friends: [
+    "Puss"
+  ]
+};
+
+const inhabitants = [dog, cat, man, woman, catWoman];
+
+inhabitants.forEach((item) => {
+  let description = "";
+  for (let key in item) {
```

Comment:
Just replace unobvious iteration across properties with an excplicit list of properties, map them into values from current object (with a special handling of nested arrays just like you did in code fragment below), and join the result with a delimeter.

Code will become cleaner and more obvious.

Reply:
Do you mind some like this:
inhabitants.forEach((item) => {
  print(`${item.species}; ${item.name}; ${item.gender}`);
});
But I wanted to make an universal solution that could print all the properties from all the objects.
Ok, I can remake it.

Reply:
This is better. It will print exactly the props we want. This makes us safe adding props that we may not want to print.
It refers properties explicitly which is helpful when you revert to the code back after centuries and you can **quickly** understand what happens here **exactly**.

Also there is a pattern here: `` `${item.species}; ${item.name}; ${item.gender}` ``
Can you make this part DRYer? E.g. employing `Array` methods.

Reply:
I think I can make it more DRY if I convert objects to arrays and use some array method for all props (like .map()). But in this case I will not be able to select which props to print and which not to print, and in fact this will become my old code but with "Array.map()" method instead of loop "for (let key in object)".
Sorry, but I don't understand which way you want me to go - list all props "by hands" or use automation, but some specific automation, not the one I used before.

Reply:
I remade it with ability of choosing props to print and without repeating any pattern parts.

Reply:
Thank for hints in AMA session :)
I hope I understood you right.